src/parameters/types.ts

```diff
@@ -83,6 +86,11 @@ type IntegerParameterDefinition = BaseParameterDefinition<number> & {
   choices?: EnumChoice<number>[];
 };
 
+export type OAuth2ParameterDefinition = BaseParameterDefinition<string> & {
+  type: typeof SlackTypes.oauth2;
+  oauth2_provider_key: string;
```

Comment:
I was hoping that there'd be a way to enforce that this key is always present when the type of an input is `SlackTypes.oauth2`. I experimented with a few changes but wasn't able to get it to work. Fortunately `apps.manifest.validate` does this check via JSON Schema and `slack deploy` errors out with the following error:

> 1: Unexpected schema encountered for `slack#/types/credential/oauth2` type: missing required field: oauth2_provider_key (missing_field)
Source: /functions/reverse/input_parameters/googleCredential